comm/src/test/scala/coop/rchain/kademlia/distance.scala

```diff
@@ -0,0 +1,121 @@
+package coop.rchain.kademlia
+
+import org.scalatest._
+import scala.util.{Success, Try}
+import scala.concurrent.duration.{Duration,MILLISECONDS}
+
+case class PeerNode(val key: Seq[Byte]) extends Peer {
+  val rand = new scala.util.Random
+
+  override def ping: Try[Duration] = {
+    Success(Duration(100, MILLISECONDS))
+  }
+
+  override def toString = {
+    val skey = key.map("%02x" format _).mkString
+    s"#{PeerNode $skey}"
+  }
+}
+
+object b {
+  val rand = new scala.util.Random(System.currentTimeMillis)
+  def apply(i: Int): Byte = i.asInstanceOf[Byte]
+  def apply(s: String): Byte = b(Integer.parseInt(s, 2))
+  def rand(nbytes: Int): Seq[Byte] = {
+    val arr = new Array[Byte](nbytes)
+    rand.nextBytes(arr)
+    arr
+  }
+}
+
+class DistanceSpec extends FlatSpec with Matchers {
+  "A PeerNode of width n byte" should "have distance to itself equal to 8n" in {
+    for (i <- 1 to 64) {
+      val home = PeerNode(b.rand(i))
+      val nt = PeerTable(home)
+      nt.distance(home) should be (Some(8*nt.width))
+    }
+  }
+
+  val width = 32
+
+  // Make 8*width copies all of which differ in a single, distinct bit
+  def oneOffs(key: Seq[Byte]): Seq[Array[Byte]] = {
+    for (i <- 0 to width - 1; j <- 7 to 0 by -1) yield {
```

Comment:
We can replace `i <- 0 to width - 1` with `i <- 0 until width`, I think it's increases readability a little bit.